src/code/z_play.c

```diff
@@ -1,59 +1,1464 @@
 #include <ultra64.h>
 #include <global.h>
+#include <vt.h>
+
+void* D_8012D1F0 = NULL;
+UNK_TYPE D_8012D1F4 = 0; // unused
+Input* D_8012D1F8 = NULL;
+
+TransitionStruct D_801613B0;
+s32 D_80161490;
+VisMonoStruct D_80161498;
+Color_RGBA8 D_801614B0;
+FaultClient D_801614B8;
+s16 D_801614C8;
+u64 D_801614D0[0xA00];
 
 void func_800BC450(GlobalContext* globalCtx) {
-    func_8005A7A8(globalCtx->cameraCtx.activeCameraPtrs[globalCtx->cameraCtx.unk_5C0], globalCtx->unk_1242B - 1,
-                  globalCtx);
+    func_8005A7A8(globalCtx->cameraPtrs[globalCtx->activeCamera], globalCtx->unk_1242B - 1, globalCtx);
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/func_800BC490.s")
+void func_800BC490(GlobalContext* globalCtx, s16 point) {
+    if (!(point == 1 || point == 2)) {
+        __assert("point == 1 || point == 2", "../z_play.c", 2160);
+    }
+
+    globalCtx->unk_1242B = point;
+
+    if ((YREG(15) != 0x10) && (gSaveContext.cutsceneIndex < 0xFFF0)) {
+        Audio_PlaySoundGeneral((point == 1) ? NA_SE_SY_CAMERA_ZOOM_DOWN : NA_SE_SY_CAMERA_ZOOM_UP, &D_801333D4, 4,
+                               &D_801333E0, &D_801333E0, &D_801333E8);
+    }
+
+    func_800BC450(globalCtx);
+}
 
 s32 func_800BC56C(GlobalContext* globalCtx, s16 arg1) {
-    return arg1 == globalCtx->unk_1242B;
+    return (arg1 == globalCtx->unk_1242B);
 }
 
+// original name: "Game_play_shop_pr_vr_switch_set"
 void func_800BC590(GlobalContext* globalCtx) {
```

Comment:
Although we have the original name for this one, I still have no idea what it does so I think it's better to leave it unnamed for now.

Reply:
I just looked in the game as to what this global context variable is. Its set to 0 normally. It becomes 1 when you enter a shop, and then gets set to 2 when you interact with the shop owner. This points the camera at their face in a first person view). Possibly what the VR comes from? lmao

This variable makes very slight changes to the camera overall. I may post a video in the discord to show what I mean lol

Reply:
heres what I mean https://youtu.be/xtHAs2BPRXs
not enough to name it right now imo since there still more to learn about the camera related variable. But if I had to go with something right now it would be like Gameplay_SetShoppingCamera or something lol. 2 Is used specifically when youre actually shopping with what testing I did

Reply:
Interesting, thanks for looking into it. The game seems to call it a "viewpoint" but yeah I don't think we have enough to name it yet.

Reply:
agreed